### PR TITLE
Updating dependencies regarding issues 721 715 714 691

### PR DESCRIPTION
**Pull request by PaurushGarg**
**Description:** <Describe what has changed.>

This PR is to resolve issue https://github.com/aws-observability/aws-otel-test-framework/issues/736.
Completed are https://github.com/aws-observability/aws-otel-test-framework/pull/721, https://github.com/aws-observability/aws-otel-test-framework/pull/715, https://github.com/aws-observability/aws-otel-test-framework/pull/691, https://github.com/aws-observability/aws-otel-test-framework/pull/714.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: load-generator/build.gradle [hunk 1]
```diff
@@ -32,9 +32,10 @@ dependencies {
 
     implementation platform("io.opentelemetry:opentelemetry-bom:1.12.0")
     implementation "io.opentelemetry:opentelemetry-api"
-    implementation "io.opentelemetry:opentelemetry-sdk-metrics:1.4.1-alpha"
-    implementation "io.opentelemetry:opentelemetry-exporter-otlp:1.10.0"
-    implementation "io.opentelemetry:opentelemetry-exporter-otlp-metrics:1.4.1-alpha"
+    implementation "io.opentelemetry:opentelemetry-sdk-metrics:1.18.0"
```
Comment (rapphil) [inline load-generator/build.gradle]: Since you are use a bom, you don't need to define version for the dependencies:

https://blog.mrhaki.com/2019/04/gradle-goodness-use-bill-of-materials.html

You can just update the version of the bom: 
```
   implementation platform("io.opentelemetry:opentelemetry-bom:1.18.0")
   implementation "io.opentelemetry:opentelemetry-api"
   implementation "io.opentelemetry:opentelemetry-sdk-metrics"
....
```

File: load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/OtlpMetricEmitter.java [hunk 2]
```diff
@@ -47,43 +53,46 @@ public void emitDataLoad() throws Exception {
...
+                    .setInstrumentationVersion("1.0")
+                    .build();
+
+    apiBytesSentCounter =
+            meter.counterBuilder(API_COUNTER_METRIC)
+                    .setDescription("API request load sent in bytes")
+                    .setUnit("one")
+                    .build();
+
+    meter.gaugeBuilder(API_LATENCY_METRIC)
```
Comment (rapphil) [inline load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/OtlpMetricEmitter.java:87]: Maybe you can use a histogram to record latency instead of a gauge: https://aws-otel.github.io/docs/getting-started/java-sdk/trace-manual-instr#creating-metrics

File: load-generator/build.gradle (hunk 1 above)
Comment (PaurushGarg) [reply load-generator/build.gradle]: Thanks. updated accordingly.

File: load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/OtlpMetricEmitter.java (hunk 2 above)
Comment (PaurushGarg) [reply load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/OtlpMetricEmitter.java:87]: thanks. as discussed keeping Gauge for now.

File: load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/OtlpMetricEmitter.java [hunk 3]
```diff
@@ -47,43 +53,46 @@ public void emitDataLoad() throws Exception {
 
   @Override
   public void nextDataPoint() {
-    apiLatencyRecorder.record(200, Labels
-        .of(DIMENSION_API_NAME, API_NAME, DIMENSION_STATUS_CODE, "200"));
-
     apiBytesSentCounter
-        .add(100, Labels.of(DIMENSION_API_NAME, API_NAME, DIMENSION_STATUS_CODE, "200"));
+            .add(10, METRIC_ATTRIBUTES);
```
Comment (bryan-aguilar) [inline load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/OtlpMetricEmitter.java:57]:
```suggestion
            .add(10, METRIC_ATTRIBUTES);
```
Why did you change the value being added? Will this affect the tests in any way?

**Review by bryan-aguilar — APPROVED** [review]

**Review by rapphil — APPROVED** [review]
